File: shared/DriftCore/Sources/DriftCore/Models/Conversation.swift
import Foundation

public struct Conversation: Codable, Equatable, Identifiable, Sendable {
    public let id: UUID
    public let profileAId: UUID
    public let profileBId: UUID
    public var tone: Tone
    public var lastReadA: Date?
    public var lastReadB: Date?
    public var mutedByA: Bool
    public var mutedByB: Bool
    public let createdAt: Date

    public init(
        id: UUID = UUID(),
        profileAId: UUID,
        profileBId: UUID,
        tone: Tone = .slow,
        lastReadA: Date? = nil,
        lastReadB: Date? = nil,
        mutedByA: Bool = false,
        mutedByB: Bool = false,
        createdAt: Date = Date()
    ) {
        self.id = id
        self.profileAId = profileAId
        self.profileBId = profileBId
        self.tone = tone
        self.lastReadA = lastReadA
        self.lastReadB = lastReadB
        self.mutedByA = mutedByA
        self.mutedByB = mutedByB
        self.createdAt = createdAt
    }

    public var profileIds: [UUID] { [profileAId, profileBId] }
}

public struct Message: Codable, Equatable, Identifiable, Sendable {
    public let id: UUID
    public let conversationId: UUID
    public let authorId: UUID
    public let text: String
    public let createdAt: Date

    public init(
        id: UUID = UUID(),
        conversationId: UUID,
        authorId: UUID,
        text: String,
        createdAt: Date = Date()
    ) {
        self.id = id
        self.conversationId = conversationId
        self.authorId = authorId
        self.text = text
        self.createdAt = createdAt
    }
}

/// Returned by the `reply-suggest` Edge Function. Never persisted.
public struct ReplySuggestion: Codable, Equatable, Sendable {
    public let casual: String
    public let context: String
    public let playful: String
    public let tone: Tone

    public init(casual: String, context: String, playful: String, tone: Tone) {
        self.casual = casual
        self.context = context
        self.playful = playful
        self.tone = tone
    }
}

File: apps/DriftDate/android/core/src/main/java/com/americangroupllc/drift/core/models/Conversation.kt
package com.americangroupllc.drift.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Conversation(
    val id: String,
    @SerialName("profile_a_id") val profileAId: String,
    @SerialName("profile_b_id") val profileBId: String,
    val tone: Tone = Tone.SLOW,
    @SerialName("last_read_a")  val lastReadA: String? = null,
    @SerialName("last_read_b")  val lastReadB: String? = null,
    @SerialName("muted_by_a")   val mutedByA: Boolean = false,
    @SerialName("muted_by_b")   val mutedByB: Boolean = false,
    @SerialName("created_at")   val createdAt: String,
) {
    val profileIds: List<String> get() = listOf(profileAId, profileBId)
}

@Serializable
data class Message(
    val id: String,
    @SerialName("conversation_id") val conversationId: String,
    @SerialName("author_id") val authorId: String,
    val text: String,
    @SerialName("created_at") val createdAt: Long,         // epoch millis
)

@Serializable
data class ReplySuggestion(
    val casual: String,
    val context: String,
    val playful: String,
    val tone: Tone,
)
